[executorch] generation.py with kv cache

lucylq opened the pull request.

larryliu0820 reviewed (approved):
Lint? Also does this work with our current model? Llama2 and stories

lucylq committed:
generation.py with kv cache (#3030)

Summary:
python e2e generation, using tiktoken tokenizer.

using text_completion, haven't tried chat_completion.


Test Plan:
Imported from GitHub, without a `Test Plan:` line.

Command, with prompt "Hello, I am" and seq_len = 10
```
python -m examples.models.llama2.runner.generation --pte llama_4ckpts_x.pte --tokenizer tokenizer.model --prompt="Hello I am"  --temperature=0 --params ../llama-models/llama3/params_less.json --max_gen_len=10
```

fp32, xnn, kv
fp32, xnn
same results:
```
Result: [{'generation': ' a 25 year old woman. I am a'}]
```

fp32, xnn, int4
```
Result: [{'generation': ' interested in the following products: - 1 x'}]
```

fp32, xnn, kv, sdpa (need investigation)
```
Result: [{'generation': 'ฉopteraenthalenthalenthalenthalenthalenthalenthalenthal'}]
```

Reviewed By: larryliu0820

Differential Revision: D56087430

Pulled By: lucylq
